src/main/java/sysc4806/graduateAdmissions/model/Term.java

```diff
@@ -0,0 +1,45 @@
+package sysc4806.graduateAdmissions.model;
+
+import lombok.*;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+
+/**
+ * Represents a Term (4 month period)
+ * Used to Store Deadlines and denote which Term is active
+ *
+ * @author Eric
+ */
+@Setter
+@Getter
+@NoArgsConstructor
+@ToString
+@EqualsAndHashCode
```

Comment:
This should be changed into just two equivalent annotations: 
````java
@NoArgsConstructor
@Data
````

Reply:
the @Data annotation contains a @RequiredArgsConstructor which makes a constructor with a param for the id. Since the id is auto-generated, I didn't think we would want a constructor that sets it.

Reply:
Having a constructor with an auto generated ID as a field is okay. You're right that we don't want to set it when persisting this to the database (we have to be mindful not to set) when we create a new entry, but sometimes we do want to set that ID on a new instance when a user wants to issue a PUT (idempotent update) on that entry.

As an example:
REST controller receives a PUT request with the json of {id: 1, name: "myName"} as a DTO.
We create a new copy of the entity with the fields from the DTO.
We send that off to the repository via the interface that extends JPARepository<> which will update the entry instead of completely replacing it.

I'd be cool with any combination or all of the following:
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(of = "myPrimaryKeyIdVariableName")
@Entity(name = "myOptionalNameInTheDatabaseIfI'mStoredThere")

Reply:
alright ill change it to @Data, should i still leave the constructor without the id or should that be removed?

Reply:
I would say you're fine to leave the constructor if you aren't doing the builder annotation, but if you have the builder annotation the extra constructor isn't necessary